src/content/en/updates/2018/01/paintapi.md

```diff
@@ -0,0 +1,281 @@
+width and the height of the canvas that is at our disposal. `properties` I will
+explain later in this article.
+
+Note: A paint worklet’s context is not 100% the same as a `<canvas>` context. As
+of now, text rendering methods are missing and for security reasons you cannot
+read back pixels from the canvas.
+
+As an introductory example, let’s write a checkerboard paint worklet and use it
+as a background image of a `<textarea>`. (I am using a textarea because it’s
+resizable by default.):
+
+```
+<!-- index.html -->
+<!doctype html>
+<style>
+  textarea {
+    background-image: paint(checkerboard);
+  }
+</style>
+<textarea></textarea>
+<script>
+  CSS.paintWorklet.addModule('checkerboard.js');
+</script>
+```
+
+```
+// checkerboard.js
+class CheckerboardPainter {
+  paint(ctx, geom, properties) {
+    // Use `ctx` as if it was a normal canvas
+    const colors = ['red', 'green', 'blue'];
+    const size = 32;
+    for(let y = 0; y < geom.height/size; y++) {
+      for(let x = 0; x < geom.width/size; x++) {
+        const color = colors[(x + y) % colors.length];
+        ctx.beginPath();
+        ctx.fillStyle = color;
+        ctx.rect(x * size, y * size, size, size);
+        ctx.fill();
+      }
+    }
+  }
+}
+
+// Register our class under a specific name
+registerPaint('checkerboard', CheckerboardPainter);
+```
+
+If you’ve used `<canvas>` in the past, this code should look familiar. See
+the live
+[demo](https://googlechromelabs.github.io/houdini-samples/paint-worklet/checkerboard/)
+here.
+
+Note: As with almost all new APIs, CSS Paint API is only available over HTTPS
+(or `localhost`).
+
+<img src="/web/updates/images/2018/01/paintapi/checkerboard1.png" alt="
+  Textarea with a checkerboard pattern as a background image.">
+
+The difference from using a common background image here is that the pattern
+will be re-drawn on demand, whenever the user resizes the textarea. This means
+the background image is always exactly as big as it needs to be, including the
+compensation for high-density displays.
+
+That’s pretty cool, but it’s also quite static. Would we want to write a new
+worklet every time we wanted the same pattern but with differently sized
+squares? The answer is no!
+
+### Parameterizing your worklet
+
+Luckily, the paint worklet can access other CSS properties, which is where the
+additional parameter `properties` comes into play. By giving the class a static
+`inputProperties` attribute, you can subscribe to changes to any CSS property,
+including custom properties. The values will be given to you through the
+`properties` parameter.
+
+```
+<!-- index.html -->
+<!doctype html>
+<style>
+  textarea {
+    /* The paint worklet subscribes to changes of these custom properties. */
+    --checkerboard-spacing: 10;
+    --checkerboard-size: 32;
+    background-image: paint(checkerboard);
+  }
+</style>
+<textarea></textarea>
+<script>
+  CSS.paintWorklet.addModule('checkerboard.js');
+</script>
+```
+
```

Comment:
If the linter gets cranky, you can optionally put a `<div class="clearfix"></div>` here. It's a nice little safety thing I use sometimes just to be safe.